ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+   altogether, you could not send this patch:
+
+
+
+
+
+
+
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 31]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   [[ "CalendarEvent/set", {
+     "accountId": "ue150411c",
+     "update": {
+       "123": {
+         "recurrenceOverrides/2018-03-08T09:00:00/
+             participants~1dG9tQGZvb2Jhci5xlLmNvbQ": null
+       }
+     }
+   }, "0" ]]
+
+   This would mean remove the "participants/dG9tQGZvb2Jhci5xlLmNvbQ"
+   property at path "recurrenceOverrides" -> "2018-03-08T09:00:00"
+   inside the object; but this doesn't exist.  We actually want to add
+   this property and make it map to null.  The client must instead send
+   the full object that contains the property mapping to null, like so:
+
+   [[ "CalendarEvent/set", {
+     "accountId": "ue150411c",
+     "update": {
+       "123": {
+         "recurrenceOverrides/2018-03-08T09:00:00": {
+           "start": "2018-03-08T10:00:00",
+           "participants/em9lQGZvb2GFtcGxlLmNvbQ/participationStatus":
+                                                          "declined",
+           "participants/dG9tQGZvb2Jhci5xlLmNvbQ": null
+         }
+       }
+     }
+   }, "0" ]]
+
+5.9.  CalendarEvent/copy
+
+   This is a standard "/copy" method as described in Section 5.4 of
+   [RFC8620].
+
+5.10.  CalendarEvent/query
+
+   This is a standard "/query" method as described in Section 5.5 of
+   [RFC8620], with two extra arguments:
+
+   *  *expandRecurrences*: Boolean (default: false)
+      If true, the server will expand any recurring event.  If true, the
+      filter MUST be just a FilterCondition (not a FilterOperator) and
+      MUST include both a "before" and "after" property.  This ensures
+      the server is not asked to return an infinite number of results.
+   *  *timeZone*: TimeZoneId
+      The time zone for before/after filter conditions (default: "Etc/
+      UTC")
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 32]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   If expandRecurrences is true, a separate id will be returned for each
+   instance of a recurring event that matches the query.  This synthetic
+   id is opaque to the client, but allows the server to resolve the id +
+   recurrence id for "/get" and "/set" operations.  Otherwise, a single
+   id will be returned for matching recurring events that represents the
+   entire event.
+
+   There is no necessary correspondence between the ids of different
+   instances of the same expanded event.
+
+   The following additional error may be returned instead of the
+   "CalendarEvent/query" response:
+
+   cannotCalculateOccurrences: The server cannot expand a recurrence
+   required to return the results for this query.
+
+5.10.1.  Filtering
+
+   A *FilterCondition* object has the following properties:
+
+   *  *inCalendars*: Id[]|null
```

Comment:
In reviewing this and the Contacts draft, it seems weird (but maybe fine/justifiable) that this type of argument for various types is:
 * inMailbox (In just this mailbox)
 * inAddressBook (In just this addressbook)
 * inCalendars (in any of these calendars)

Why does the calendar spec let you list multiple items but the others don't? Maybe this maps to common use cases of how these things are searched...

Reply:
Yeah, I think we should be consistent here: I've changed it to `inCalendar` to match the semantics of the other specs.